### ISSUE TITLE
docker run --net=host deletes ipv6 addresses from host interfaces

### ISSUE THREAD

**Issue by euank** [docker/docker#32639]
**Description**

Running `docker run --net=host` with docker 17.04 results in my ipv6 addresses for all of my network interfaces being lost.

**Steps to reproduce the issue:**
```shell
esk@multivac ~ $ ip a show wlp4s0  
3: wlp4s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether 10:02:b5:b2:b2:22 brd ff:ff:ff:ff:ff:ff
    inet x.y.z.101/22 brd x.y.z.255 scope global dynamic wlp4s0
       valid_lft 86328sec preferred_lft 86328sec
    inet6 fe80::246a:a46d:b875:2b5a/64 scope link 
       valid_lft forever preferred_lft forever
esk@multivac ~ $ ip a show lo    
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever


$ docker run --net=host busybox


esk@multivac ~ $ ip a show wlp4s0
3: wlp4s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether 10:02:b5:b2:b2:22 brd ff:ff:ff:ff:ff:ff
    inet x.y.z.101/22 brd x.y.z.255 scope global dynamic wlp4s0
       valid_lft 86192sec preferred_lft 86192sec
esk@multivac ~ $ ip a show lo
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever

```

**Describe the results you received:**

`lo` and `wlp4s0` no longer have inet6 addresses

**Describe the results you expected:**

No impact on my existing interfaces

**Output of `docker version`:**

```
$ docker version
Client:
 Version:      17.04.0-ce
 API version:  1.28
 Go version:   go1.7.5
 Git commit:   4845c56
 Built:        Mon Apr  3 18:06:13 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.04.0-ce
 API version:  1.28 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   4845c56
 Built:        Mon Apr  3 18:06:13 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**

```
$ docker info
Containers: 27
 Running: 0
 Paused: 0
 Stopped: 27
Images: 71
Server Version: 17.04.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: 
containerd version: 422e31ce907fd9c3833a38d7b8fdd023e5a76e73
runc version: 9c2d8d184e5da67c95d601382adf14862e4f2228
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
 selinux
Kernel Version: 4.9.9-200.fc25.x86_64
Operating System: Fedora 25 (Workstation Edition)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.678GiB
Name: multivac.euank.com
ID: 753V:GKN7:7TOV:KJUP:ELNG:Q45G:4HRT:TVWD:U2WL:6ZNQ:G4BT:YUQV
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Username: euank
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

This is running on a fedora25 machine with it installed normally from the distro repositories.

**Comment by euank** [comment]
More information:

```shell
$ head /proc/sys/net/ipv6/conf/*/disable_ipv6
==> /proc/sys/net/ipv6/conf/all/disable_ipv6 <==
0

==> /proc/sys/net/ipv6/conf/default/disable_ipv6 <==
0

==> /proc/sys/net/ipv6/conf/docker0/disable_ipv6 <==
0

==> /proc/sys/net/ipv6/conf/enp0s25/disable_ipv6 <==
0

==> /proc/sys/net/ipv6/conf/lo/disable_ipv6 <==
0

==> /proc/sys/net/ipv6/conf/virbr0/disable_ipv6 <==
1

==> /proc/sys/net/ipv6/conf/virbr0-nic/disable_ipv6 <==
0

==> /proc/sys/net/ipv6/conf/virbr1/disable_ipv6 <==
1

==> /proc/sys/net/ipv6/conf/virbr1-nic/disable_ipv6 <==
0

==> /proc/sys/net/ipv6/conf/wlp4s0/disable_ipv6 <==
0

$ docker run --net=host busybox


$ head /proc/sys/net/ipv6/conf/*/disable_ipv6
==> /proc/sys/net/ipv6/conf/all/disable_ipv6 <==
1

==> /proc/sys/net/ipv6/conf/default/disable_ipv6 <==
1

==> /proc/sys/net/ipv6/conf/docker0/disable_ipv6 <==
1

==> /proc/sys/net/ipv6/conf/enp0s25/disable_ipv6 <==
1

==> /proc/sys/net/ipv6/conf/lo/disable_ipv6 <==
1

==> /proc/sys/net/ipv6/conf/virbr0/disable_ipv6 <==
1

==> /proc/sys/net/ipv6/conf/virbr0-nic/disable_ipv6 <==
1

==> /proc/sys/net/ipv6/conf/virbr1/disable_ipv6 <==
1

==> /proc/sys/net/ipv6/conf/virbr1-nic/disable_ipv6 <==
1

==> /proc/sys/net/ipv6/conf/wlp4s0/disable_ipv6 <==
1

**Comment by thaJeztah** [comment]
I think this is a duplicate of https://github.com/docker/docker/issues/32433, and fixed in 17.05 through https://github.com/docker/docker/pull/32447

**Comment by euank** [comment]
Indeed, thanks!
